components/core/src/clp/GlobalMetadataDBConfig.cpp

```diff
@@ -1,110 +1,99 @@
+    } else if ("mysql" == db_type_string) {
+        metadata_db_type = GlobalMetadataDBConfig::MetadataDBType::MySQL;
+    } else {
+        throw invalid_argument("Unknown database type: " + db_type_string);
+    }
 
-static exception
-get_yaml_unconvertable_value_exception(string const& key_name, string const& destination_type) {
-    throw invalid_argument(
-            fmt::format("'{}' could not be converted to type '{}'", key_name, destination_type)
-    );
+    return in;
 }
 
-namespace clp {
-void GlobalMetadataDBConfig::parse_config_file(string const& config_file_path) {
-    YAML::Node config = YAML::LoadFile(config_file_path);
+auto GlobalMetadataDBConfig::add_command_line_options(
+        boost::program_options::options_description& options_description
+) -> void {
+    options_description.add_options()
+            ("db-type",
+             boost::program_options::value<MetadataDBType>(&m_metadata_db_type)->default_value(cDefaultMetadataDbType, cDefaultMetadataDbTypeName.data()),
+             "Database type [sqlite | mysql]")
+            ("db-host",
+             boost::program_options::value<string>(&m_metadata_db_host)->default_value(cDefaultMetadataDbHost.data()),
+             "[db-type=mysql] Database host")
+            ("db-port",
+             boost::program_options::value<int>(&m_metadata_db_port)->default_value(cDefaultMetadataDbPort),
+             "[db-type=mysql] Database port")
+            ("db-name",
+             boost::program_options::value<string>(&m_metadata_db_name)->default_value(cDefaultMetadataDbName.data()),
+             "[db-type=mysql] Database name")
+            ("db-table-prefix",
+             boost::program_options::value<string>(&m_metadata_table_prefix)->default_value(cDefaultMetadataTablePrefix.data()),
+             "[db-type=mysql] Database table prefix");
+}
 
-    if (!config["type"]) {
-        throw get_yaml_missing_key_exception("type");
+auto GlobalMetadataDBConfig::read_credentials_from_env() -> void {
+    auto const* db_username{std::getenv("CLP_DB_USER")};
+    if (nullptr != db_username) {
+        m_metadata_db_username = db_username;
     }
 
-    auto db_type_string = config["type"].as<string>();
-    if ("sqlite" == db_type_string) {
-        m_metadata_db_type = MetadataDBType::SQLite;
-    } else if ("mysql" == db_type_string) {
-        m_metadata_db_type = MetadataDBType::MySQL;
+    auto const* db_password{std::getenv("CLP_DB_PASS")};
+    if (nullptr != db_password) {
+        m_metadata_db_password = db_password;
+    }
+}
 
-        if (!config["host"]) {
-            throw get_yaml_missing_key_exception("host");
-        }
-        try {
-            m_metadata_db_host = config["host"].as<string>();
-        } catch (YAML::BadConversion& e) {
-            throw get_yaml_unconvertable_value_exception("host", "string");
-        }
+auto GlobalMetadataDBConfig::validate() const -> void {
+    if (m_metadata_db_type == MetadataDBType::MySQL) {
```

Comment:
shall we validate that those parameters are not provided when db_type is not mysql? (unfortunately, it seems the only way to check "not provided" is to compare them with the default value.)

It can be helpful to give user an error when they specifying parameters that are not going to have effect. When we work with someone from another company, there was a time when someone provided all paramaters of a fature but missing an `--enable-feature` flag. The user just told us "the feature doesn't work" and It took us some cycle to finally understand the issue